blocks/editable/format-toolbar.js

```diff
@@ -63,17 +63,19 @@ class FormatToolbar extends Component {
 	}
 
 	componentWillReceiveProps( nextProps ) {
-		const newState = {
-			linkValue: nextProps.formats.link ? nextProps.formats.link.value : '',
-		};
+		// Update the link value if the focused link node changes
 		if (
-			! this.props.formats.link ||
-			! nextProps.formats.link ||
-			this.props.formats.link.node !== nextProps.formats.link.node
+			!! nextProps.formats.link !== !! this.props.formats.link ||
```

Comment:
Ok does something like this looks good?

`isUndefined( a ) !== isUndefined( b )`